### DISCUSSION TITLE
Why does the "row" class does not force divs to a row?

### DISCUSSION THREAD

**Discussion by wsw70** [quasarframework/quasar#10175] [General - Components / Directives / etc]
I have a piece of code where two `div` should be aligned in a row:

      <div class="row">   
        <div class="col col-shrink">  
          <div class="text-bold">{{ extractTime(e.start) }}</div>
          <div class="text-h5" v-if="isLongerThanAnHour(e)">
            → {{ extractTime(e.end) }}
          </div>
        </div>
        <div class="q-ml-md">{{ e.name }}</div>  
      </div>

The effect is unfortunately

![image](https://user-images.githubusercontent.com/1345886/126830875-47060190-7b72-44f0-8f87-69d79275962a.png)

The red arrow shows that the class has been applied correctly.

I was expecting the `div` with the hours to be in the same row as the text, but they are stacked. What could be the reason for that?

*Note: I am at the "arrangement" phase so please do not judge the aesthetics of the page yet :)*

**Comment by hawkeye64** [comment]
@wsw70 Not sure if this will help, but make it `row no-wrap`. The other thing is, use padding instead of margins (then likely don't need `no-wrap`).

**Reply by wsw70** [reply]
Thank you - it indeed helped. I am not sure why it wrapped but the view with `no-wrap` is a bit weird

![image](https://user-images.githubusercontent.com/1345886/126838037-8eaaff57-4eb4-4a13-857d-7768a3f9bd2e.png)

I think this will be because of the margin, as you suggest. I will make some tests and update this comment if I have a good enough result (I ported the components from an older, pure-Vue version, and I am rearranging and tweaking the Quasar version)

**Reply by hawkeye64** [reply]
If you are going to use margin, then wrap that with a parent div

**Reply by hawkeye64** [reply]
Another thing you can try is `row q-gutter-sm`, but remove all margin and padding on children first

**Reply by wsw70** [reply]
> Another thing you can try is `row q-gutter-sm`

Ah, that's a good idea. I am using this home dashboard app I developped a few years ago to get used to Quasar and its built-in CSS  helpers (as opposed to specific styling). I am still pondering how much of these CSS should be Quasar-based, and how much in `<style>` (I am an amateur dev)
